lib/publiccloud/provider.pm

```diff
@@ -444,18 +448,23 @@ sub terraform_destroy {
     my $cmd;
     return if get_required_var('TEST') =~ /^publiccloud_upload_img/;
     record_info('INFO', 'Removing terraform plan...');
+
+    # Add region variable also to `terraform destroy` (poo#63604) -- needed by AWS.
     if (get_var('PUBLIC_CLOUD_SLES4SAP')) {
         assert_script_run('cd ' . TERRAFORM_DIR . '/' . $self->conv_openqa_tf_name);
-        $cmd = 'terraform destroy -no-color -auto-approve';
+
+        # On HA/SAP tests with ha-sap-terrafrom-deployments tool we shouldn't force the region for all providers
+        $cmd = sprintf(q(terraform destroy -no-color -auto-approve -var 'aws_region=%s'), $self->region);
```

Comment:
Isn't there a way to avoid `sprintf` here?

Reply:
There is, but why?

Reply:
Readability? 
I'm just prejudiced against `sprintf` for being insecure in C, but Perl is not C, so feel free to ignore my suggestion.

Reply:
`$cmd = "terraform destroy -no-color -auto-approve -var aws_region=" . $self->region` should work.

Reply:
@cfconrad Which syntax do you prefer?

Reply:
I would say it is really matter of taste , I personally like more version @ricardobranco777  but I think you can decided on your own if you want to change it or not . It is pure "eye candy" :)

Reply:
As this line was with a `sprintf` before, and as there is other `sprintf` commands in this file I prefer to keep it as is.

@ricardobranco777 there are some chances that the next change will be from you, so feel free to change it next time! ;-)

Reply:
@ricardobranco777 , @ldevulder , @cfconrad  WDYT about such version ? 
```
$cmd = 'terraform destroy -no-color -auto-approve';
$cmd .= ' -var 'aws_region=' . $self->region if is_ec2();
```

Note : you will need to export `is_ec2()` from https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/lib/publiccloud/utils.pm

Reply:
In that case it's better to do this I think: `$cmd .= " -var 'aws_region=$self->region'" if is_ec2;`, as the single quote is also used for the `aws_region`, and protected all is more complicated IMHO.

I'm doing a last change and I will re-trigger jobs.

Reply:
Verification runs are OK (again Azure could fail, because of another not related bug).